packages/app/src/cli/services/info.ts

```diff
@@ -117,157 +122,165 @@ class AppInfo {
           const {name, roles} = web.configuration
-          sublevels.push([`    📂 ${name} (${roles.join(',')})`, relativePath(this.app.directory, web.directory)])
+          const pathToWeb = relativePath(this.app.directory, web.directory)
+          sublevels.push([`    📂 ${name}`, {filePath: pathToWeb || '/'}])
+          if (roles.length > 0) {
+            sublevels.push(['         roles', roles.join(', ')])
+          }
         } else {
           web.configuration.roles.forEach((role) => {
-            sublevels.push([`    📂 ${role}`, relativePath(this.app.directory, web.directory)])
+            sublevels.push([`    📂 ${role}`, {filePath: relativePath(this.app.directory, web.directory)}])
           })
         }
       } else {
-        sublevels.push([`  📂 ${UNKNOWN_TEXT}`, relativePath(this.app.directory, web.directory)])
+        sublevels.push([{subdued: `  📂 ${UNKNOWN_TEXT}`}, {filePath: relativePath(this.app.directory, web.directory)}])
       }
       if (this.app.errors) {
         const error = this.app.errors.getError(`${web.directory}/${configurationFileNames.web}`)
         if (error) errors.push(error)
       }
     })
-    let errorContent = `\n${errors.map((error) => this.formattedError(error)).join('\n')}`
-    if (errorContent.trim() === '') errorContent = ''
 
-    return `${subtitle}\n${linesToColumns([toplevel, ...sublevels])}${errorContent}`
+    return this.subtableSection('web', [
+      ['📂 web', ''],
+      ...sublevels,
+      ...errors.map((error): InlineToken[] => [{error: 'error'}, {error: this.formattedError(error)}]),
+    ])
   }
 
-  extensionSubSection(extension: ExtensionInstance): string {
+  extensionsSections(): CustomSection[] {
+    const extensions = this.app.allExtensions.filter((ext) => ext.isReturnedAsInfo())
+    const types = Array.from(new Set(extensions.map((ext) => ext.type)))
+    return types
+      .map((extensionType: string): CustomSection | undefined => {
+        const relevantExtensions = extensions.filter((extension: ExtensionInstance) => extension.type === extensionType)
+        if (relevantExtensions[0]) {
+          return this.subtableSection(
+            relevantExtensions[0].externalType,
+            relevantExtensions.map((ext) => this.extensionSubSection(ext)).flat(),
+          )
+        }
+      })
+      .filter((section: CustomSection | undefined) => section !== undefined)
+  }
+
+  extensionSubSection(extension: ExtensionInstance): InlineToken[][] {
     const config = extension.configuration
-    const details = [
-      [`📂 ${extension.handle}`, relativePath(this.app.directory, extension.directory)],
-      ['     config file', relativePath(extension.directory, extension.configurationPath)],
+    const details: InlineToken[][] = [
+      [`📂 ${extension.handle || NOT_LOADED_TEXT}`, {filePath: relativePath(this.app.directory, extension.directory)}],
+      ['     config file', {filePath: relativePath(extension.directory, extension.configurationPath)}],
```

Comment:
this is probably unnecessary, since the config file is always `shopify.extension.toml`. 

Should we remove it?